.github/workflows/blocked-prs.yml

```diff
@@ -0,0 +1,245 @@
+name: Blocked/Stacked Pull Requests Automation
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - reopened
+      - edited
+      - synchronize
+  workflow_dispatch:
+    inputs:
+      pr_id:
+        description: Local Pull Request number to work on
+        required: true
+        type: number
+
+jobs:
+  blocked_status:
+    name: Check Blocked Status
+    runs-on: ubuntu-latest
+
+    permissions:
+      issues: write
+      pull-requests: write
+      statuses: write
+      checks: write
+
+    steps:
+      - name: Checkout Default Branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.repository.default_branch }}
+      - name: Setup From Pull Request Event
+        if: github.event_name != 'workflow_dispatch'
+        id: pr_event_setup
+        env:
+          PR_JSON: "${{ toJSON(github.event.pull_request) }}"
+        run: |
+          # setup env for the rest of the workflow
+          {
+            echo "REPO=$(jq -r '.base.repo.name' <<< "$PR_JSON")"
+            echo "OWNER=$(jq -r '.base.repo.owner.login' <<< "$PR_JSON")"
+            echo "PR_NUMBER=$(jq -r '.number' <<< "$PR_JSON")"
+            echo "JOB_DATA=$(jq -c '
+              {
+                "repo": .base.repo.name,
+                "owner": .base.repo.owner.login,
+                "repoUrl": .base.repo.html_url,
+                "prNumber": .number,
+                "prHeadSha": .head.sha,
+                "prHeadLabel": .head.label,
+                "prBody": .body,
+                "prLabels": (reduce .labels[].name as $l ([]; . + [$l]))
+              }
+            ' <<< "$PR_JSON")"
+          } >> "$GITHUB_ENV"
+
+      - name: Setup From Dispatch Event
+        if: github.event_name == 'workflow_dispatch'
+        id: dispatch_event_setup
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPOSITORY: ${{ github.repository }}
+          PR_NUMBER: ${{ inputs.pr_id }}
+        run: |
+          # setup env for the rest of the workflow
+          OWNER=$(echo "$REPOSITORY" | cut -d '/' -f 1)
+          REPO=$(echo "$REPOSITORY" | cut -d '/' -f 2)
+          PR_JSON=$(
+            gh api \
+              -H "Accept: application/vnd.github.raw+json" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              "/repos/$OWNER/$REPO/pulls/$PR_NUMBER"
+          )
```

Comment:
This section could be moved to it's own step before both of these

Reply:
ok I finally caught on to what you ment